packages/xdl/src/start/ExpoUpdatesManifestHandler.ts

```diff
@@ -29,14 +36,80 @@ function getPlatformFromRequest(req: express.Request | http.IncomingMessage): st
   return String(platform);
 }
 
+async function shouldUseAnonymousManifestAsync(): Promise<boolean> {
+  const currentSession = await UserManager.getSessionAsync();
+  return !currentSession || ConnectionStatus.isOffline();
+}
+
+async function getScopeKeyAndProjectIdForExpoConfigAsync(
+  expoConfig: ExpoConfig
+): Promise<{
+  scopeKey: string;
+  projectId: string | null;
+}> {
+  const shouldUseAnonymousManifest = await shouldUseAnonymousManifestAsync();
+  if (shouldUseAnonymousManifest) {
+    const userAnonymousIdentifier = await UserSettings.getAnonymousIdentifierAsync();
+    return {
+      scopeKey: `@${ANONYMOUS_USERNAME}/${expoConfig.slug}-${userAnonymousIdentifier}`,
+      projectId: null,
+    };
+  }
+
+  const user = await UserManager.ensureLoggedInAsync();
+  const username = await UserManager.getCurrentUsernameAsync();
```

Comment:
Is there no user.username field?

Reply:
There is but `getCurrentUsernameAsync` gets the username from a different place, so it's very confusing.

Reply:
```suggestion
  const [user, username] = await Promise.all([UserManager.ensureLoggedInAsync(), UserManager.getCurrentUsernameAsync()]);
```